internal/page/donor/upload_evidence.go

```diff
@@ -51,29 +72,33 @@ func UploadEvidence(tmpl template.Template, payer Payer, donorStore DonorStore,
 			data.Errors = form.Validate()
 
 			if data.Errors.None() {
-				for _, file := range form.Files {
-					uuid := randomUUID()
-					key := lpa.UID + "-evidence-" + uuid
-
-					_, err := s3Client.PutObject(r.Context(), &s3.PutObjectInput{
-						Bucket:               aws.String(evidenceBucketName),
-						Key:                  aws.String(key),
-						Body:                 bytes.NewReader(file.Data),
-						ServerSideEncryption: types.ServerSideEncryptionAwsKms,
-						Tagging:              aws.String("replicate=true"),
-					})
-					if err != nil {
-						return err
+				if form.Action == "upload" {
+					for _, file := range form.Files {
+						uuid := randomUUID()
+						key := lpa.UID + "-evidence-" + uuid
```

Comment:
You might want to put a / rather than a - so that the bucket is easier to diagnose issues in when it's got a few more items in.

Reply:
to add to this comment, / would be used to represent folder structures whereas - would appear as all objects in the root of a folder representation

https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html